yaml-sort.js

```diff
@@ -88,28 +93,56 @@ argv.input.forEach((file) => {
     }
 
     const output =
-        argv.stdout || (argv.output === '.') || (isStdin && !argv.output)
-          ? process.stdout.fd
-          : (argv.output ? argv.output : file)
+      argv.stdout || (argv.output === '.') || (isStdin && !argv.output)
+        ? process.stdout.fd
+        : (argv.output ? argv.output : file)
 
     const content = fs.readFileSync(isStdin ? process.stdin.fd : file, argv.encoding)
+    let documents
+    try {
+      documents = yaml.loadAll(content)
+    } catch (error) {
+      success = false
+      if (process.env?.DEBUG === 'true') {
+        console.error(error)
+      } else {
+        console.error(error?.reason)
+      }
+    }
+    if (!documents) {
+      process.exit(1)
+    }
 
-    const sorted = yaml.dump(yaml.load(content), {
+    const sortedDocuments = documents.map(doc => yaml.dump(doc, {
       sortKeys: true,
       indent: argv.indent,
       lineWidth: argv.lineWidth,
-      quotingType: argv.quotingStyle === 'double' ? '"' : "'"
-    })
+      quotingType: argv.quotingStyle === 'double' ? '"' : "'",
+      forceQuotes: argv.forceQuotes
+    }))
+
+    const hasDocumentStart = content.toString().trimStart().startsWith('---')
+
+    const sortedContent = (hasDocumentStart ? '---\n' : '') + sortedDocuments.join('---\n')
 
     if (argv.check) {
-      if (sorted !== content.toString()) {
+      const originalContent = (hasDocumentStart ? '---\n' : '') + documents
+        .map(doc => yaml.dump(doc, {
+          indent: argv.indent,
+          lineWidth: argv.lineWidth,
+          quotingType: argv.quotingStyle === 'double' ? '"' : "'",
+          forceQuotes: argv.forceQuotes
+        }))
+        .join('---\n')
+
+      if (sortedContent !== originalContent) {
```

Comment:
@Romain7495 Why doesn't it check `sortedContent` with `content.toString()` like before?

Reply:
Hello @ddebin, there is an edge case, when you have a key in you yaml `y: foo`, `js-yaml` lib will wrap it into `'y'` because `y` is ambiguous. If we compare with `toString`, it will compare `'y': foo` with `y: foo`

I have added two test cases that verify this in `test/test-multiple-sorted.yml` and `test/test-multiple.yml` files

Reply:
@Romain7495 
Replaced `if (sortedContent !== originalContent)` with `if (sortedContent !== content.toString())` and it still passes tests...

Reply:
well ok, it's fixed